### PR TITLE
noetic-devel

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: hmi_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,10 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.0.2)
 project(hmi_msgs)
 
 find_package(catkin REQUIRED COMPONENTS
   actionlib_msgs
   message_generation
+  std_msgs
```
Comment (jlunenburg) [inline hmi_msgs/CMakeLists.txt:7]: Why do we need this dependency?

File: hmi/src/hmi/common.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def result_to_ros(result):
 
 
 def result_from_ros(msg):
-    return HMIResult(sentence=msg.sentence, semantics=json.loads(msg.semantics))
+    return HMIResult(sentence=msg.sentence, semantics=str(json.loads(msg.semantics)))
```
Comment (jlunenburg) [inline hmi/src/hmi/common.py:20]: Not sure why this is necessary?

**Review by jlunenburg — COMMENTED** [review]
Added some questions

File: hmi/src/hmi/common.py (hunk 2 above)
Comment (MatthijsBurgh) [reply hmi/src/hmi/common.py:20]: Unicode vs bytes issues

File: hmi_msgs/CMakeLists.txt (hunk 1 above)
Comment (MatthijsBurgh) [reply hmi_msgs/CMakeLists.txt:7]: That is the default

File: hmi/src/hmi/common.py (hunk 2 above)
Comment (jlunenburg) [reply hmi/src/hmi/common.py:20]: So basically, it's because of Python2 - Python3 compatibility.

**Pull request merged**